core/src/test/java/org/apache/iceberg/MetadataTableScanTestBase.java

```diff
@@ -89,18 +88,16 @@ protected void validatePartition(
       CloseableIterable<ManifestEntry<? extends ContentFile<?>>> entries,
       int position,
       int partitionValue) {
-    Assert.assertTrue(
-        "File scan tasks do not include correct file",
-        StreamSupport.stream(entries.spliterator(), false)
-            .anyMatch(
-                entry -> {
-                  StructLike partition = entry.file().partition();
-                  if (position >= partition.size()) {
-                    return false;
-                  }
-
-                  return Objects.equals(partitionValue, partition.get(position, Object.class));
-                }));
+    assertThat(StreamSupport.stream(entries.spliterator(), false))
```

Comment:
```suggestion
    assertThat(entries)
```
I think this should also work